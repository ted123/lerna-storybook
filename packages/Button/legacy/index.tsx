import * as React from "react";
import * as ReactDOM from "react-dom";

import Button from "../src/index";

ReactDOM.render(
  <Button compiler="TypeScript" framework="React" />,
  document.getElementById("Button")
);
