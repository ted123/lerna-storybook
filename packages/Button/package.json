{
  "name": "@4ward/testbutton",
  "description": "test component",
  "version": "0.0.6",
  "main": "lib/index.js",
  "license": "MIT",
  "scripts": {
    "tsc": "tsc",
    "legacy": "webpack --config legacy/webpack.config.js"
  },
  "gitHead": "ad8f6db0a55694d10fd7c8bcdf801855e1ab4d90"
}
