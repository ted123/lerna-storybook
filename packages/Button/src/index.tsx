import * as React from "react";
import styled from "styled-components";

export interface ButtonProps {
  compiler: string;
  framework: string;
}

const Wrapper = styled.div`
  border: 1px solid blue;
  padding: 10px;
`;

const Button = (props: ButtonProps) => (
  <Wrapper>
    <h1>
      Hello World from {props.compiler} and {props.framework}!
    </h1>
  </Wrapper>
);

export default Button;
