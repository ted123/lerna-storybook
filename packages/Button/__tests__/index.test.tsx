import * as React from "react";
import { shallow } from "enzyme";
import Button from "../src/index";

describe("Button package tests", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<Button compiler="TypeScript" framework="React" />);
    expect(wrapper).toMatchSnapshot();
  });
});
