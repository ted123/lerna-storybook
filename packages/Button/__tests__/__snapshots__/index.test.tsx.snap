// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button package tests matches snapshot 1`] = `
<styled.div>
  <h1>
    Hello World from 
    TypeScript
     and 
    React
    !
  </h1>
</styled.div>
`;
